### PR TITLE
Pull requests

### PR DESCRIPTION
**Pull request by wangtzh**

### CODE REVIEW

File: foedus-core/include/foedus/assorted/zipfian_random.hpp [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ *
+ * HP designates this particular file as subject to the "Classpath" exception
+ * as provided by HP in the LICENSE.txt file that accompanied this code.
+ */
+#ifndef FOEDUS_ASSORTED_ZIPFIAN_RANDOM_HPP_
+#define FOEDUS_ASSORTED_ZIPFIAN_RANDOM_HPP_
+
+#include <stdint.h>
+
+#include "uniform_random.hpp"
```
Comment (hkimura) [inline foedus-core/include/foedus/assorted/zipfian_random.hpp]: must be always full path.

File: foedus-core/include/foedus/assorted/zipfian_random.hpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * The major user is YCSB. 0 < theta < 1, lower means more skewed.
+ * Generates a random number between 0 and max_exclusive_ -1.
+ * @ingroup ASSORTED
+ */
+class ZipfianRandom {
+ private:
+  double zeta(uint64_t n) {
+    double sum = 0;
+    for (uint64_t i = 0; i < n; i++) {
+      sum += 1 / std::pow(i + 1, theta_);
```
Comment (hkimura) [inline foedus-core/include/foedus/assorted/zipfian_random.hpp]: include cmath

File: foedus-core/include/foedus/assorted/zipfian_random.hpp [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+ public:
+  explicit ZipfianRandom(uint64_t items, double theta, uint64_t urnd_seed) {
+    init(items, theta, urnd_seed);
+  }
+
+  explicit ZipfianRandom() {}
+
+  uint64_t next() {
+    double u = urnd_.uniform_within(0, 100) / 100.0;
```
Comment (hkimura) [inline foedus-core/include/foedus/assorted/zipfian_random.hpp]: is it 100 inclusive? both params in uniform_within are inclusive

**Comment by hkimura** [comment]
I want to see a unit test for zipfian, but it can be another PR. just fix the include path pls

**Comment by wangtzh** [comment]
Is there a way to verify the numbers, other than only printing them out? I already have a unittest already does that.

**Comment by hkimura** [comment]
You mean, to report as failure if the numbers are incorrect or exceed reasonable range? EXPECT_EQ/LT etc are for that. Because this is a random generator, the testcase needs to tolerate a reasonable deviation. hashinate test does a very similar thing.

**Comment by wangtzh** [comment]
hmm, not sure about the best range for highly scewed case, the low skew case seems ok. saw printed high-skew values, looks fine...

**Pull request merged**